api/v1beta1/common_types.go

```diff
@@ -101,6 +101,83 @@ type BarbicanComponentTemplate struct {
 	NetworkAttachments []string `json:"networkAttachments,omitempty"`
 }
 
+// BarbicanLunaTemplate - Variables specific to Thales' Luna HSMs
+type BarbicanLunaTemplate struct {
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default=4
+        // +kubebuilder:validation:Maximum=7
+        // +kubebuilder:validation:Minimum=0
+        // Level of logging, where 0 means "no logging" and 7 means "debug".
+        LunaLoggingLevel int `json:"lunaLoggingLevel"`
+
+        // +kubebuilder:validation:Optional
+        // The HSM certificates. The map's key is the HSM's IP address and
+        // the value is the OpenShift secret storing the certificate.
+        LunaHSMCertificates map[string]string `json:"lunaHsmCertificates"`
+
+        // +kubebuilder:validation:Optional
+        // The OpenShift secret storing the client certificate plus its key
+        LunaClientCertificate string `json:"lunaClientCertificate"`
+
```

Comment:
If I recall correctly, we've decided that we're going to take these files and put them into a secret which is to be mounted.  So that means that what you really need here is a parameter that contains the name of the secret to be mounted.

Reply:
Actually, now that I think about it, you're going to need the secret - but also you're going to need the path names for the certs.

Reply:
Kindly observe the comment right above the variable name. It's not meant to directly provide the certificate, but actually the secret that contains the certificate.